mkosi/__init__.py

```diff
@@ -325,6 +327,39 @@ def script_maybe_chroot(script: Path, mountpoint: str) -> list[str]:
     return ["mkosi-chroot", mountpoint] if script.suffix == ".chroot" else [os.fspath(script)]
 
 
+@contextlib.contextmanager
+def finalize_scripts(scripts: Mapping[str, Sequence[PathString]] = {}) -> Iterator[Path]:
+    with tempfile.TemporaryDirectory(prefix="mkosi-scripts") as d:
+        for name, script in scripts.items():
+            # Make sure we don't end up in a recursive loop when we name a script after the binary it execs
+            # by removing the scripts directory from the PATH when we execute a script.
+            (Path(d) / name).write_text(
+                textwrap.dedent(
+                    f"""\
+                    #!/bin/sh
+                    DIR="$(cd "$(dirname "$0")" && pwd)"
+                    PATH="$(echo $PATH | tr ':' '\\n' | grep -v $DIR | tr '\\n' ':')"
```

Comment:
```suggestion
                    PATH="$(echo "$PATH" | tr ':' '\\n' | grep -v "$DIR" | tr '\\n' ':')"
```
Let's be safe here, those are user directories, they might well have spaces or other shenanigans in them.